docs/adr/2023-11-07_029-streaming-persistence.md

```diff
@@ -0,0 +1,40 @@
+---
+slug: 29
+title: |
+  29. EventServer abstraction for event persistence
+authors: [@cardenaso11, @quantumplation]
+tags: []
+---
+
+## Status
+Draft
+
+## Context
+
+The Hydra node represents a significant engineering asset, providing layer 1 monitoring, peer to peer consensus, durable persistence, and an isomorphic Cardano ledger. Because of this, it is being eyed as a key building block not just in Hydra based applications, but other protocols as well.
+
+One remaining difficulty in integrating Hydra into a fully productionalized software stack is the persistence model.  Currently, the Hydra node achieves durability by writing a sequence of "StateChanged" events to disk. If a node is interrupted, upon restarting, it can replay just these events, ignoring their corresponding effects, to recover the same internal state it had when it was interrupted. However, downstream consumers don't have the same luxury.
+
+We propose generalizing the persistence mechanism to open the door to a plugin based approach to extending the Hydra node.
```

Comment:
This sentence seems more relevant to the `Decision` section. I would suggest replacing it with concrete examples. 
eg. 
```
For example, a hydra-node deployed as part of a larger system on k8s could want store it's event stream 
using some persistence service afforded by the stack it is part of instead of plain, "naive", disk-based 
persistence. This would allow scenarios like graceful shutdown and restart across redeployments, or 
redundancy of hydra-node-based services through a primary-secondary architecture.
```